java-frontend/src/test/files/se/StreamNotConsumedCheck.java

```diff
@@ -97,6 +97,32 @@ void exceptionalPath3(Stream<Object> stream) {
     stream.forEach(System.out::println);
   }
 
+  boolean useMethodInvocation1(List<Boolean> list) {
+    Stream<Boolean> filter = list.stream().filter(Boolean::booleanValue); // Compliant
+    return consumes(filter::iterator);
+  }
+
+  Stream<Boolean> field;
+  boolean useMethodInvocation2(List<Boolean> list) {
+    this.field = list.stream().filter(Boolean::booleanValue); // Compliant
+    return consumes(this.field::iterator);
+  }
+
+  boolean useMethodInvocation3(List<Boolean> list) {
+    Stream<Boolean> filter = list.stream().filter(Boolean::booleanValue); // Compliant
+    uses(filter);
+    return consumes(filter::iterator);
+  }
+
+  boolean useMethodInvocation4(List<Boolean> list) {
+    Stream<Boolean> filter = list.stream().filter(Boolean::booleanValue); // Compliant
+    java.util.Iterator<Boolean> itr = filter.iterator();
+    return consumes(filter::iterator);
```

Comment:
This should raise issue on S3959  because stream is already consumed. This works when lines are switched, and call using method reference is before the actual invocation.